tools/checkdocs/checkdocs.go

```diff
@@ -38,10 +40,92 @@ func main() {
 		log.Fatal(err)
 	}
 
+	docsFiles, err := getMarkdownFiles(filepath.Join("website", "docs"))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	p, err := github.CacheFilePath()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	client, err := github.NewClient(p, log.Printf, gh.NoopPrintf, gh.NoopPrintf)
+	if err != nil {
+		log.Fatal(err)
+	}
+
 	err = checkBlogFiles(blogFiles)
 	if err != nil {
 		log.Fatal(err)
 	}
+
+	err = checkDocFiles(client, docsFiles)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// checkDocFiles verifies docs contain valid issue URLs.
+func checkDocFiles(client *github.Client, files []string) error {
+	var failed bool
+
+	for _, file := range files {
+		f, err := os.Open(file)
+		if err != nil {
+			return fmt.Errorf("could not read file %s: %s", file, err)
+		}
+
+		defer func() {
+			if err = f.Close(); err != nil {
+				log.Fatal(err)
+			}
+		}()
+
+		issueURLFailed, err := checkIssueURLs(client, f, file, log.Default())
+		if err != nil {
+			log.Printf("%q: %s", file, err)
+			failed = true
+		}
+
+		if issueURLFailed {
+			failed = true
+		}
+	}
+
+	if failed {
+		return fmt.Errorf("one or more docs contain invalid issue URLs")
+	}
+
+	return nil
+}
+
+// getMarkdownFiles returns markdown files in the given directory and 3 levels of subdirectories.
+func getMarkdownFiles(path string) ([]string, error) {
+	// https://github.com/golang/go/issues/11862
+	paths := []string{
+		filepath.Join(path, "*.md"),
+		filepath.Join(path, "*.mdx"),
+		filepath.Join(path, "*", "*.md"),
+		filepath.Join(path, "*", "*.mdx"),
+		filepath.Join(path, "*", "*", "*.md"),
+		filepath.Join(path, "*", "*", "*.mdx"),
+		filepath.Join(path, "*", "*", "*", "*.md"),
+		filepath.Join(path, "*", "*", "*", "*.mdx"),
+	}
```

Comment:
That will break when we change the structure. Let's use `filepath.Walk` instead.